draft-irtf-cfrg-hpke.md

```diff
@@ -958,6 +1001,21 @@ For X25519 and X448, public keys and Diffie-Hellman outputs MUST be validated
 as described in {{RFC7748}}. In particular, recipients MUST check whether
 the Diffie-Hellman shared secret is the all-zero value and abort if so.
 
+### Future KEMs {#future-kems}
+
+{{kem-security}} lists security requirements on a KEM used within HPKE.
+
+The `AuthEncap` and `AuthDecap` functions are OPTIONAL. If a KEM algorithm
+does not provide them, only the Base and PSK modes of HPKE are supported.
+Future specifications which define new KEMs MUST indicate whether or not
+these modes are supported.
+
+A KEM algorithm MAY define multiple different equivalent encoding
+algorithms for KEM public keys or KEM ciphertexts, with different output
+lengths. In this case `Npk` and `Nenc` MUST indicate the maximum output
+length. It is left to the application using HPKE to define how entities
+communicate which encoding is used.
```

Comment:
Having multiple different equivalent encodings seems like a bad idea.   Why are we accommodating this?

Reply:
I think this is just a presentation issue. It may be the case that future KEMs output encapsulated keys with non-fixed-length. So we can either (a) require that those KEMs normalize on a fixed length, or (b) acknowledge that they may be variable length here, right?

Reply:
We are accommodating this because of this review comment we got (https://github.com/cfrg/draft-irtf-cfrg-hpke/issues/84):

```
Assumptions about the length of the public key. It may not always be a
fixed value, "Npk", for a KEM with a given set of parameters. The other
(and unrelated) "hybrid" draft, draft-ietf-tls-hybrid-design, Section 3.2,
makes accommodation for public keys associated with a given set of
parameters to vary in size.
```

Here is a quote from Section 3.2 from https://tools.ietf.org/html/draft-ietf-tls-hybrid-design-00
```
for example, the SIKE key encapsulation mechanism permits
compressed or uncompressed public keys at each security level, and
the compressed and uncompressed formats are interoperable.
```

Reply:
I have no strong opinion on if we should require that those KEMs normalize on a fixed length or if we should just acknowledge the variable lengths. It might be beneficial to leave more freedom to future KEMs. I agree though that the current “maximum length” formulations seem a bit messy.